mautic/mautic
# Updates Zoho to their REST v2 API

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/7651
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Zoho is sunsetting v1 of their API tomorrow. This migrates to v2 of their API. It requires re-authorizing with them. 

A portion of this code was provided by Korcomptenz. 

[//]: # ( As applicable: )
#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. In Mautic, go to the Plugins page and click on Zoho
3. Copy the callback URL
4. In Zoho, register a new client https://www.zoho.com/crm/developer/docs/api/register-client.html to generate a client ID and secret. Use the callback URL copied from Mautic
5. After saving the client in Zoho, click the vertical ellipsis to the right of the new client and click Edit
6. Copy the Client ID to Mautic's Zoho config for "Application Client ID" and copy the Client Secret to "Application Client Secret"
7. Click the Authorize or Reauthorize button
8. A popup will open requiring you to login with your Zoho credentials then click Allow to authorize Mautic's integration
9. Map contact and/or company fields with priority facing Mautic
10. Enable all the features and select Leads, Contacts, and Accounts as objects to sync
11. Create a campaign with a "Push to Integration" action configured for Zoho and add contacts to the campaign
12. Execute the campaign and check that the Mautic contacts were created as Leads in Zoho
13. Now create a few new contacts in Mautic and different ones in Zoho as Leads and others as Contacts
14. Let the sync execute (run `php app/console mautic:integration:synccontacts -i Zoho`)
15. Check that the new contacts in Mautic were synced to Zoho as leads and that the Zoho Leads and Contacts were synced to Mautic
16. Edit the name of one of the contacts in Mautic and sync again. It should not have synced to Zoho
17. Edit one of the Leads or Contacts in Zoho and sync. The update should have synced to Mautic.
18. In Mautic, check the profiles of a contact that was created in Mautic and one that was pushed to Zoho. There should be an entry in the Integrations tab for Zoho with the Zoho ID. 
19. Create an Account in Zoho and it should sync back to Mautic (the existing Zoho plugin does not currently support pushing companies from Mautic to Zoho and this PR does not implement it).
